api/v1alpha1/ingressnodefirewall_webhook.go

```diff
@@ -17,24 +17,97 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"fmt"
+	"net"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
 	ctrl "sigs.k8s.io/controller-runtime"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
+//This constant is defined in C code used to generate LPM trie map value array
+//TODO(mk): Move this constant to a better location and import it
+const MAX_INGRESS_RULES = 100
+
+type empty struct{}
+type uint32Set map[uint32]empty
+
+type pinHole interface {
+	isRuleConflict(rule IngressNodeFirewallProtocolRule) (bool, error)
+}
+
+type transportProtoPinHole struct {
+	serviceName string
+	port        uint16
+}
+
+func (t transportProtoPinHole) isRuleConflict(rule IngressNodeFirewallProtocolRule) (bool, error) {
+	if rule.ProtocolRule == nil {
+		return false, fmt.Errorf("rule is not defined")
+	}
+	if rule.ProtocolRule.IsRange() {
+		start, end, err := rule.ProtocolRule.GetRange()
+		if err != nil {
+			return false, err
+		}
+		return withinRange(t.port, start, end), fmt.Errorf("port range is in conflict with access to %s", t.serviceName)
+	} else {
+		port, err := rule.ProtocolRule.GetPort()
+		if err != nil {
+			return false, err
+		}
+		return t.port == port, fmt.Errorf("port is in conflict with access to %s", t.serviceName)
+	}
+}
+
+func withinRange(i, lowerBound, upperBound uint16) bool {
+	if lowerBound > upperBound {
+		panic("Expected lower number to be less than or equal upper number")
```

Comment:
why panic instead of returning and error and fail the config ?

Reply:
Fixed. Thanks. Moved outside this func.